stdlib/socket/from.go

```diff
@@ -0,0 +1,309 @@
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/memory"
+	"github.com/influxdata/flux/plan"
+	"github.com/influxdata/flux/semantic"
+	"github.com/influxdata/flux/values"
+	"github.com/pkg/errors"
+)
+
+const FromSocketKind = "fromSocket"
+
+type RawDecoder struct {
+	reader *bufio.Reader
+	stats  flux.Statistics
+	a      execute.Administration
+}
+
+func (rd *RawDecoder) getCurrentTime() execute.Time {
+	return rd.a.ResolveTime(flux.Time{
+		IsRelative: false,
+		Absolute:   time.Now(),
+	})
+}
+
+func (rd *RawDecoder) Do(f func(flux.Table) error) error {
+	vs := make([]string, 0)
+	times := make([]values.Time, 0)
+	var start, stop values.Time
+	var eof bool
+
+	// Get every record and keep it in memory.
+	// We cannot create the table yet, because we don't know the values
+	// of the group key (i.e., start and stop).
+	start = rd.getCurrentTime()
+	for !eof {
+		s, err := rd.reader.ReadString('\n')
+		if err != nil && err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+
+		vs = append(vs, strings.Trim(s, "\n "))
+		times = append(times, rd.getCurrentTime())
+	}
+	stop = rd.getCurrentTime()
+
+	startCol := flux.ColMeta{Label: "_start", Type: flux.TTime}
+	stopCol := flux.ColMeta{Label: "_stop", Type: flux.TTime}
+	timeCol := flux.ColMeta{Label: "_time", Type: flux.TTime}
+	valueCol := flux.ColMeta{Label: "_value", Type: flux.TString}
+
+	ks := []flux.ColMeta{startCol, stopCol}
+	key := execute.NewGroupKey(ks, []values.Value{values.New(start), values.New(stop)})
+	builder := execute.NewColListTableBuilder(key, &memory.Allocator{})
+	startIdx, err := builder.AddCol(startCol)
+	if err != nil {
+		return err
+	}
+	stopIdx, err := builder.AddCol(stopCol)
+	if err != nil {
+		return err
+	}
+	timeIdx, err := builder.AddCol(timeCol)
+	if err != nil {
+		return err
+	}
+	valueIdx, err := builder.AddCol(valueCol)
+	if err != nil {
+		return err
+	}
+
+	for i := 0; i < len(vs); i++ {
+		err = builder.AppendTime(startIdx, start)
+		if err != nil {
+			return err
+		}
+		err = builder.AppendTime(stopIdx, stop)
+		if err != nil {
+			return err
+		}
+		err = builder.AppendTime(timeIdx, times[i])
+		if err != nil {
+			return err
+		}
+		err = builder.AppendString(valueIdx, vs[i])
+		if err != nil {
+			return err
+		}
+	}
+
+	tbl, err := builder.Table()
+	if err != nil {
+		return err
+	}
+
+	rd.stats = tbl.Statistics()
+
+	return f(tbl)
+}
+
+func (*RawDecoder) Name() string {
+	return "_result"
+}
+
+func (rd *RawDecoder) Tables() flux.TableIterator {
+	return rd
+}
+
+func (rd *RawDecoder) Statistics() flux.Statistics {
+	return rd.stats
+}
+
+func (rd *RawDecoder) Decode(r io.Reader) (flux.Result, error) {
+	rd.reader = bufio.NewReader(r)
+	return rd, nil
+}
+
+var decoders = map[string]bool{
+	"json": false, // TODO(affo) should flatten json into table
+	"csv":  true,
+	"raw":  true,
+}
+
+type FromSocketOpSpec struct {
+	URL     string `json:"url"`
+	Decoder string `json:"decoder"`
+}
+
+func init() {
```

Comment:
Can you move this init function to the top of the file?  That way its easy to see the function signature being used and follows the convention of the rest of the builtins.

Reply:
Sure 👍